### PR TITLE
Create a 3.x release branch

### PR DESCRIPTION
**Pull request by RangerRick**
This pr creates `release-3.x` and includes some fixes to deployment to make sure release/non-`SNAPSHOT` maven deploys happen (including GPG jar signing).

### CODE REVIEW

File: DEVEL.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ Run the Karaf container:
 
 In the Karaf shell, add the features repository:
 ```
-feature:repo-add mvn:org.opennms.alec/alec-karaf-features/3.0.0-SNAPSHOT/xml
+feature:repo-add mvn:org.opennms.alec/alec-karaf-features/3.0.0/xml
```
Comment (BenjaminJ) [inline DEVEL.md]: 3.0.1-SNAPSHOT ?

File: UI.md [hunk 2]
```diff
@@ -43,8 +43,8 @@ admin@opennms()>
 
 In OpenNMS Karaf shell:
 ```
-admin@opennms()> feature:repo-add mvn:org.opennms.alec/alec-karaf-features/3.0.0-SNAPSHOT/xml                                                                                                          
-Adding feature url mvn:org.opennms.alec/alec-karaf-features/3.0.0-SNAPSHOT/xml
+admin@opennms()> feature:repo-add mvn:org.opennms.alec/alec-karaf-features/3.0.0/xml                                                                                                          
+Adding feature url mvn:org.opennms.alec/alec-karaf-features/3.0.0/xml
```
Comment (BenjaminJ) [inline UI.md]: 3.0.1-SNAPSHOT ?

**Comment by RangerRick** [comment]
Ergh, good catch. We should probably make those things be filtered/generated by maven instead of needing hand-editing. (Well, maybe not `DEVEL.md`, but the _code_ that has versions frozen in it.)

I didn't realize `mvn versions:set` was not sufficient.

**Review by BenjaminJ — COMMENTED** [review]
LGTM

**Pull request merged**